ndi/src/lib.rs

```diff
@@ -471,6 +471,29 @@ impl VideoData {
         }
     }
 
+    /// creata VideoData from a raw memory buffer, with custom settings
+    pub fn from_buffer(width: i32, height: i32, fourcc: FourCCVideoType, framerate : i32, frame_format: FrameFormatType, timecode: i64, stride:i32, buffer: &mut [u8] ) -> Self {
+        Self {
+            p_instance: NDIlib_video_frame_v2_t {
+                xres: width,
+                yres: height,
+                FourCC: fourcc as _,
+                frame_rate_N: framerate,
+                frame_rate_D: 0,
```

Comment:
The frame_rate_N and frame_rate_D stand for numerator / denominator. Therefore, in this case I think `frame_rate_D = 1` is the thing you intend to write.  I tested this locally and setting `framerate: 30` came back with a video_data which had framerate 59.966 or something else. Changing the denominator to 1 resolved that issue.